prototype_source/backend_config_tutorial.rst

```diff
@@ -0,0 +1,400 @@
+  (scale and zero point) and ranges when quantizing to a given data
+  type. Each DTypeConfig consists of many of these.
+
+The pattern specified in BackendPatternConfig follows the format
+described `here <https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/backend_config/README.md#pattern-specification>`__.
+
+BackendPatternConfig Specification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+set_observation_type
+^^^^^^^^^^^^^^^^^^^^
+
+Observation type here refers to how observers (or QuantDeQuantStubs) are
+placed in the graph. There are two observation types:
+
+- ``OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT`` (default): the output
+  observer instance will be different from the input. This is the most
+  common observation type.
+- ``OUTPUT_SHARE_OBSERVER_WITH_INPUT``: the output observer instance
+  will be the same as the input. This is useful for operators like ``cat``.
+
+Note: This will be renamed in the near future, since we will soon insert
+QuantDeQuantStubs with observers (and fake quantizes) attached instead
+of observers themselves.
+
+set_dtype_configs / add_type_config
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Each operator pattern may support one or more sets of
+input/output/weight/bias data types, and each set may have their own
+constraints. These requirements are captured in DTypeConfigs, which will
+be described in more detail in the next section.
+
+set_root_module / set_reference_quantized_module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When we construct the reference quantized model during the convert
+phase, the root modules (e.g. ``torch.nn.Linear`` for
+``torch.ao.nn.intrinsic.LinearReLU``) will be swapped to the
+corresponding reference quantized modules (e.g.
+``torch.ao.nn.reference.quantized.Linear``). This allows custom backends
+to specify custom reference quantized module implementations to match
+the numerics of their lowered operators. Since this is a one-to-one
+mapping, both the root module and the reference quantized module must be
+specified in the same BackendPatternConfig in order for the conversion
+to take place.
+
+set_fuser_method
+^^^^^^^^^^^^^^^^
+
+As an optimization, operator patterns such as (``torch.nn.Linear``,
+``torch.nn.ReLU``) may be fused into ``nni.LinearReLU``.
+``set_fuser_method`` specifies the function through which this is
+performed. The first argument of this function is ``is_qat``, and the
+rest of the arguments are the items in the tuple pattern, e.g. the fuser
+method for the above pattern will have three arguments, ``is_qat``,
+``linear``, and ``relu``. See `this
+example <https://gist.github.com/jerryzh168/8bea7180a8ba3c279f2c9b050f2a69a6>`__
+for a slightly more complicated usage.
+
+set_fused_module
+^^^^^^^^^^^^^^^^
+
+This is used to identify fused weighted modules (e.g.
+``torch.ao.nn.intrinsic.LinearReLU``) that need to be converted to
+reference quantized modules.
+
+Data Type Restrictions
```

Comment:
This still reads like "interface_dtype".  I feel like this section should have an example with an expected reference model, and an explicit mention that the dtype here is the dtype argument to the quantize function inside the reference model.

Check out `observer.py` which is more explicit in defining what dtype means inside of it.

Reply:
Ok, I'll fix the wording here too to be more explicit